vtex/faststore
# feat: add nextjs redirects

## What's the purpose of this pull request?

This PR enables users to define [Next.js redirects](https://nextjs.org/docs/pages/api-reference/next-config-js/redirects).

## How it works?

Define the redirects function in the `faststore.config.js`  file. If you need to import files with redirect definitions, put them inside `src/` and require them by using the `BASE_PATH` property received via params. It is not possible to import files using other methods yet.

Don't define the redirect function like the [Next.js doc](https://nextjs.org/docs/pages/api-reference/next-config-js/redirects) without explicitly defining the property name. Use the `propertyName: value` format.

## How to test it?

Browse the deploy preview below and enter the url `/my-test`. You should be redirected to `/my-test-new`.

### Starters Deploy Preview

https://github.com/vtex-sites/starter.store/pull/231

## References

https://nextjs.org/docs/pages/api-reference/next-config-js/redirects
#2070
#2071
